core/primitives/src/shard_layout.rs

```diff
@@ -85,6 +111,17 @@ impl ShardLayout {
         }
     }
 
+    #[inline]
+    pub fn get_parent_shard_id(&self, shard_id: ShardId) -> Option<ShardId> {
+        match self {
+            Self::V0(_) => None,
+            Self::V1(v1) => match &v1.to_parent_shard_map {
+                Some(to_parent_shard_map) => Some(to_parent_shard_map[shard_id as usize]),
```

Comment:
Question: should we perhaps `to_parent_shard_map.get(shrd_id)?` here? As in (way out of my depth here), can someone submit us a challenge which would cause us to run this fn with large `shard_id` which is out of bounds?

Reply:
Good point, I think such it shouldn't be possible for such challenges to reach code places like this because we would check number of shards before calling this function. But yeah that's not obvious to see here. I will change it.